### ISSUE TITLE
Error: undefined variable $response in DAV storage

### ISSUE THREAD

**Issue by ChristophWurst** [nextcloud/server#11140]
### Steps to reproduce
1. Call https://github.com/nextcloud/server/blob/8601bbc35d915d6dc1c847560d148981c518c27f/lib/private/Files/Storage/DAV.php#L829
2. Hit https://github.com/nextcloud/server/blob/master/lib/private/Files/Storage/DAV.php#L855
3. Hit https://github.com/nextcloud/server/blob/8601bbc35d915d6dc1c847560d148981c518c27f/lib/private/Files/Storage/DAV.php#L283

### Expected behaviour
Condition is detected and handled (logged, ignored).

### Actual behaviour
An unexpected condition is met and a non-existent variable is accessed.

Ref https://sentry.rullzer.com/sentry/nextcloud/issues/150 (for people with access).

**Comment by ChristophWurst** [comment]
@nextcloud-bot nope, not even close …

**Comment by skjnldsv** [comment]
Still relevant @ChristophWurst ?

**Comment by kesselb** [comment]
Yes. See https://github.com/nextcloud/server/issues/16471 for example. We should not reach `return $response` if undefined.

**Comment by skjnldsv** [comment]
Nice!
So duplicate or relevant of #16471 @kesselb ?

**Comment by kesselb** [comment]
Relevant but not a duplicate. #16471 is running into the undefined return but the problem is with sabre/dav or alfresco. This issue is about the incomplete error handling.

**Comment by szaimen** [comment]
I suppose this issue is still valid? If not, please close this issue!

**Comment by szaimen** [comment]
Hi, please update to 24.0.8 or better 25.0.2 and report back if it fixes the issue. Thank you!
